cloud/amazon/sns_topic.py

```diff
@@ -228,9 +252,14 @@ def main():
         if not next_token:
             break
 
-    desired_subscriptions = [(sub['protocol'],
```

Comment:
Wouldn't it be simpler to just set the default for subscriptions to be [] instead of None? Then you wouldn't need to check for `subscriptions is not None`.